components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -75,6 +82,18 @@ public static AuthenticationContextCache getInstance() {
      * @param entry Actual object where cache entry is placed.
      */
     public void addToCache(AuthenticationContextCacheKey key, AuthenticationContextCacheEntry entry) {
+
+        if (isSessionDataStorageOptimizationEnabled && entry.getContext() != null) {
+            try {
+                AuthenticationContextLoader.getInstance().optimizeAuthenticationContext(entry.getContext());
+            } catch (SessionDataStorageOptimizationException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Error occurred while optimizing the Authentication context with " +
+                            "context id: %s", entry.getContext().getContextIdentifier()), e);
+                }
+                return;
+            }
+        }
```

Comment:
let's move this logic before ` SessionDataStore.getInstance().storeSessionData` block 

idea is, we don't need to optimize it if we are not going to persist the object.

Reply:
Fixed with [`3b214ce`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/3b214ce93851b12e42d7fc213904a8306abba29a)